src/Octoshift/SecretScanningAlertService.cs

```diff
@@ -0,0 +1,165 @@
+                continue;
+            }
+
+            await _targetGithubApi.UpdateSecretScanningAlert(targetOrg, targetRepo, target.Alert.Number,
+                alert.Alert.State, alert.Alert.Resolution);
+            _log.LogSuccess(
+                $"  source and target alert state and resolution have been aligned to {alert.Alert.Resolution}.");
+        }
+    }
+
+    private AlertWithLocations MatchTargetSecret(AlertWithLocations source, List<AlertWithLocations> targets)
+    {
+        AlertWithLocations matched = null;
+
+        foreach (var target in targets)
+        {
+            if (matched != null)
+            {
+                break;
+            }
+
+            if (source.Alert.SecretType == target.Alert.SecretType
+                && source.Alert.Secret == target.Alert.Secret)
+            {
+                _log.LogVerbose(
+                    $"Secret type and value match between source:{source.Alert.Number} and target:{source.Alert.Number}");
+                var locationMatch = true;
+                foreach (var sourceLocation in source.Locations)
+                {
+                    locationMatch = IsMatchedSecretAlertLocation(sourceLocation, target.Locations);
+                    if (!locationMatch)
+                    {
+                        break;
+                    }
+                }
+
+                if (locationMatch)
+                {
+                    matched = target;
+                }
+            }
+        }
+
+        return matched;
+    }
+
+    private bool IsMatchedSecretAlertLocation(GithubSecretScanningAlertLocation sourceLocation,
+        GithubSecretScanningAlertLocation[] targetLocations)
+    {
+        var sourceDetails = sourceLocation.Details;
+
+        // We cannot guarantee the ordering of things with the locations and the APIs, typically they would match, but cannot be sure
+        // so we need to iterate over all the targets to ensure a match
+        foreach (var targetLocation in targetLocations)
+        {
+            var targetDetails = targetLocation.Details;
+
+            if (sourceDetails.Path == targetDetails.Path
+                && sourceDetails.StartLine == targetDetails.StartLine
+                && sourceDetails.EndLine == targetDetails.EndLine
+                && sourceDetails.StartColumn == targetDetails.StartColumn
+                && sourceDetails.EndColumn == targetDetails.EndColumn
+                && sourceDetails.BlobSha == targetDetails.BlobSha)
+            // Technically this wil hold, but only if there is not commmit rewriting going on, so we need to make this last one optional for now
+            // && sourceDetails.CommitSha == targetDetails.CommitSha)
+            {
+                return true;
+            }
+        }
+
+        return false;
```

Comment:
`foreach` can be replaced by a simple LINQ method, see my [previous comment](https://github.com/github/gh-gei/pull/592#discussion_r964366163)